src/celeritas/global/detail/AlongStepActionImpl.hh

```diff
@@ -162,14 +170,19 @@ inline CELER_FUNCTION void along_step_track(CoreTrackView const& track)
                      && msc_result.step_length <= step_limit.step);
         step_limit.step = msc_result.step_length;
 
-        // Update direction and position
+        // Update direction
         geo.set_dir(msc_result.direction);
-        Real3 new_pos;
-        for (int i : range(3))
+
+        // Update position only if there is a lateral displacement
+        if (msc_step_result.is_displaced)
```

Comment:
This still isn't quite right, since some cases in the MscScatter (such as being very close to the boundary) still result in a zero displacement even when the earlier step length allows it. The change in #455 to return a new "action result" should hopefully fix the error for all occurrences.

Reply:
Okay, a good point.  Yes, it is a side effect by the refactoring, updating geo outside actual place inside the scatter.  Anyway, let's use a corresponding msc action result for position changed that you introduced (in your last MR) in a following MR.  Just to be consistent, we also pass the const reference of the geometry view (instead of the point) as we does not update geometry inside the scatter after the refactoring code.